syst/make_cutsets_multitrial.py

```diff
@@ -0,0 +1,141 @@
+        yaml.dump(cfg_flow, out_file, default_flow_style=False)
+    with open(os.path.join(f"{output_dir}_multitrial/config_history/", f"config_modifications.yml"), 'w') as out_file:
+        yaml.dump(cfg_mod, out_file, default_flow_style=False)
+
+    for _, pt_bin in enumerate(cfg_mod['ptbins']):
+        
+        ptmin = pt_bin['range'][0]
+        ptmax = pt_bin['range'][1]
+        pt_bin_index = cfg_flow['ptbins'].index(ptmin)
+        ref_config_ptbin = get_reference_config_pt_bin(cfg_flow, pt_bin_index)
+        
+        os.makedirs(f"{output_dir}_multitrial/pt_{int(ptmin*10)}_{int(ptmax*10)}/", exist_ok=True)
+        output_file = os.path.join(f"{output_dir}_multitrial/pt_{int(ptmin*10)}_{int(ptmax*10)}/", f"config_reference.yml")
+        with open(output_file, 'w') as out_file:
+            yaml.dump(ref_config_ptbin, out_file, default_flow_style=False)
+
+        varied_configs = {}
+        print(f"ptbin: {pt_bin}")
+        for var in pt_bin["multitrial"]:
+            print(f"var: {var}")
+            if var in ref_config_ptbin:
+                print("Already in reference config")
+                varied_configs[var] = list(set(ref_config_ptbin[var] + pt_bin["multitrial"][var]))
+            else:
+                print("Not in reference config")
+                varied_configs[var] = pt_bin["multitrial"][var]
+
+        # Perform itertools.product to get all combinations
+        keys = list(varied_configs.keys())
+        values = list(varied_configs.values())
+        print(f"\nvalues: {values}\n")
+        combinations = list(itertools.product(*values))
+        config_variants = [dict(zip(keys, combination)) for combination in combinations] 
+
+        for config_variant in config_variants:
+            print(f"\nconfig_variant: {config_variant}\n")
+
+        # Print results
+        for idx, variant in enumerate(config_variants):
+            cfg_variant = copy.deepcopy(ref_config_ptbin)
+            cfg_variant["iTrial"] = idx
+            cfg_variant["OriginalConfig"] = config_flow
+
+            # Set directory to preprocessed files
+            cfg_variant["outdirprep"] = ref_config_ptbin['outdir']
+            
+            # Set which operations to perform
+            cfg_variant["operations"]["preprocess_data"] = False
+            cfg_variant["operations"]["preprocess_mc"] = False
+            cfg_variant["operations"]["make_yaml"] = True
+            cfg_variant["operations"]["proj_data"] = False
+            cfg_variant["operations"]["proj_mc"] = False
+            cfg_variant["operations"]["proj_multitrial"] = True
+            
+            cfg_variant["outdir"] = f"{output_dir}_multitrial/pt_{int(ptmin*10)}_{int(ptmax*10)}/trials/"
+            cfg_variant["suffix"] = f"{idx}"
+            cfg_variant["nworkers"] = 1 # parallelization is in the bash script
+            cfg_variant["MaxChi2"] = pt_bin['MaxChi2']
```

Comment:
We have to be careful with this pt-dependent quality selection. Maybe we could add a warning in case of chi2 (signif) too high (low)?